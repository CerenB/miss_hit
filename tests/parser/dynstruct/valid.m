function test_01()
  foo.x = 0;
  foo.y = 0;

  foo.('x') = 42;
  [foo.('x')] = 42;
end

function test_02()
  foo.x = 0;
  foo.y = 0;

  [foo.('x') foo.('y')] = potato();
end
