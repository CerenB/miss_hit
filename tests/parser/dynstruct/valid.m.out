----------------------------------------------------------------------
--  Parse tree for valid.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test_01>
    Body: Sequence_Of_Statements
      Statements: Simple_Assignment_Statement
        LHS: Selection
          Prefix: Identifier <foo>
          Field: Identifier <x>
        RHS: Number_Literal <0>
      Statements: Simple_Assignment_Statement
        LHS: Selection
          Prefix: Identifier <foo>
          Field: Identifier <y>
        RHS: Number_Literal <0>
      Statements: Simple_Assignment_Statement
        LHS: Dynamic_Selection
          Prefix: Identifier <foo>
          Field: Char_Array_Literal 'x'
        RHS: Number_Literal <42>
      Statements: Simple_Assignment_Statement
        LHS: Dynamic_Selection
          Prefix: Identifier <foo>
          Field: Char_Array_Literal 'x'
        RHS: Number_Literal <42>
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test_02>
    Body: Sequence_Of_Statements
      Statements: Simple_Assignment_Statement
        LHS: Selection
          Prefix: Identifier <foo>
          Field: Identifier <x>
        RHS: Number_Literal <0>
      Statements: Simple_Assignment_Statement
        LHS: Selection
          Prefix: Identifier <foo>
          Field: Identifier <y>
        RHS: Number_Literal <0>
      Statements: Compound_Assignment_Statement
        LHS: Dynamic_Selection
          Prefix: Identifier <foo>
          Field: Char_Array_Literal 'x'
        LHS: Dynamic_Selection
          Prefix: Identifier <foo>
          Field: Char_Array_Literal 'y'
        RHS: Reference
          Name: Identifier <potato>
----------------------------------------------------------------------
In valid.m, line 1
| function test_01()
|                 ^ info: this token is not linked to the ast
In valid.m, line 1
| function test_01()
|                  ^ info: this token is not linked to the ast
In valid.m, line 2
|   foo.x = 0;
|            ^ info: this token is not linked to the ast
In valid.m, line 3
|   foo.y = 0;
|            ^ info: this token is not linked to the ast
In valid.m, line 5
|   foo.('x') = 42;
|       ^ info: this token is not linked to the ast
In valid.m, line 5
|   foo.('x') = 42;
|           ^ info: this token is not linked to the ast
In valid.m, line 5
|   foo.('x') = 42;
|                 ^ info: this token is not linked to the ast
In valid.m, line 6
|   [foo.('x')] = 42;
|   ^ info: this token is not linked to the ast
In valid.m, line 6
|   [foo.('x')] = 42;
|        ^ info: this token is not linked to the ast
In valid.m, line 6
|   [foo.('x')] = 42;
|            ^ info: this token is not linked to the ast
In valid.m, line 6
|   [foo.('x')] = 42;
|             ^ info: this token is not linked to the ast
In valid.m, line 6
|   [foo.('x')] = 42;
|                   ^ info: this token is not linked to the ast
In valid.m, line 7
| end
| ^^^ info: this token is not linked to the ast
In valid.m, line 9
| function test_02()
|                 ^ info: this token is not linked to the ast
In valid.m, line 9
| function test_02()
|                  ^ info: this token is not linked to the ast
In valid.m, line 10
|   foo.x = 0;
|            ^ info: this token is not linked to the ast
In valid.m, line 11
|   foo.y = 0;
|            ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|   ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|        ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|            ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x')<anon,> foo.('y')] = potato();
|             ^^^^^^^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|                  ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|                      ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|                       ^ info: this token is not linked to the ast
In valid.m, line 13
|   [foo.('x') foo.('y')] = potato();
|                                   ^ info: this token is not linked to the ast
In valid.m, line 14
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
